src/BuildScriptGenerator/Helpers/StaticSiteGeneratorHelper.cs

```diff
@@ -0,0 +1,112 @@
+﻿// --------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+// --------------------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.Oryx.BuildScriptGenerator.Node;
+
+namespace Microsoft.Oryx.BuildScriptGenerator
+{
+    /// <summary>
+    /// Helper class for functions around static site generators.
+    /// </summary>
+    internal static class StaticSiteGeneratorHelper
+    {
+        /// <summary>
+        /// Checks whether or not the given repository uses a static site generator.
+        /// </summary>
+        /// <param name="sourceRepo">Source repo for the application.</param>
+        /// <returns>True if the app uses a static site generator, false otherwise.</returns>
+        public static bool IsStaticSite(ISourceRepo sourceRepo)
+        {
+            return IsHugoApp(sourceRepo);
+        }
+
+        /// <summary>
+        /// Checks whether or not the given repository is a Hugo application.
+        /// </summary>
+        /// <param name="sourceRepo">Source repo for the application.</param>
+        /// <returns>True if the app is a Hugo app, false otherwise.</returns>
+        public static bool IsHugoApp(ISourceRepo sourceRepo)
```

Comment:
Seems like we can also look into environment variables to determine if it is a hugo app. We should probably check that first and then fall back to config files? https://gohugo.io/getting-started/configuration/#configure-with-environment-variables